app/Filament/Resources/GameResource/RelationManagers/AchievementSetsRelationManager.php

```diff
@@ -194,7 +194,9 @@ public function table(Table $table): Table
                         // It'll use the same achievement_set_id.
 
                         $legacySubsetGame = Game::find((int) $data['game']);
-                        $legacySubsetAchievementSet = $legacySubsetGame->gameAchievementSets()->core()->first();
+                        $legacySubsetAchievementSet = $legacySubsetGame->achievementSets()
+                            ->wherePivot('type', AchievementSetType::Core->value)
+                            ->first();
```

Comment:
Could a similar problem be happening in the Search widget? I went to Metroid Prime > Sets > Attach, typed in Metroid, and got this:
![image](https://github.com/user-attachments/assets/6e5e856b-c0b6-4794-ae94-88177b6e42bc)

Reply:
I think this is a separate ordeal.

For the specific bug addressed by the change in this branch, the user actually selected the correct game in the "Subset game" field, but the tool attached an entirely different core set.

For example, when the user tried to attach a Final Fantasy subset, a Mortal Kombat set was attached by the tool instead.

These options values are coming from:
```php
->searchable()
->options(
    Game::where('Title', 'like', "%[Subset -%")
        ->where('ConsoleID', $game->ConsoleID)
        ->where('achievements_published', '>', 0)
        ->whereDoesntHave('gameAchievementSets', function ($query) use ($attachedAchievementSetIds) {
            $query->core()->whereIn('achievement_set_id', $attachedAchievementSetIds);
        })
        ->orderBy('Title')
        ->get()
        ->mapWithKeys(function ($game) {
            return [$game->id => $game->title];
        })
        ->toArray()
```

I figured fixing this select field bug would be trivial, but it seems I don't have access to the user's input search value in either `->searchable()` or `->options()`.